### PR TITLE
fix: hash ending path resolution fails

### PR DESCRIPTION
**Pull request by maininformer**
## What/Why/How?

Currently if a `$ref` ends with `#`, meaning the document root, the code breaks because it cannot find the path. For instance:

```
schema: path/to/schema#
```
This fixes https://github.com/Redocly/redocly-cli/issues/1392

## Reference

[json schema](https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-00#idExamples)
[Issue](https://github.com/Redocly/redocly-cli/issues/1392)

## Testing

E2E test [here](https://github.com/maininformer/redocly-cli/tree/fix/hash-ending-path-resolution-fails/__tests__/bundle/reference-ending-in-hash) 

## Has code been changed?

- [ ] Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/ref-utils.ts [hunk 1]
```diff
@@ -43,7 +43,8 @@ export function escapePointer<T extends string | number>(fragment: T): T {
 }
 
 export function parseRef(ref: string): { uri: string | null; pointer: string[] } {
-  const [uri, pointer] = ref.split('#/');
+  const [uri, pointer_] = ref.split('#');
```
Comment (tatomyr) [inline packages/core/src/ref-utils.ts:46]: I don't like the naming though. @malis42 could you suggest a better solution?

**Review by tatomyr — COMMENTED** [review]
Thank you @maininformer for your contribution and the original issue! The solution looks correct, but I'd like to make it a bit nicer.

File: packages/core/src/ref-utils.ts (hunk 1 above)
Comment (malis42) [reply packages/core/src/ref-utils.ts:46]: I think we should do a small refactor in this area and reuse already existing function which is defined below the `parseRef` function, like this:

```ts
export function parseRef(ref: string): { uri: string | null; pointer: string[] } {
  const [uri, pointer = ''] = ref.split('#');

  return {
    uri: uri || null,
    pointer: parsePointer(pointer),
  };
}

export function parsePointer(pointer: string) {
  return pointer.split('/').map(unescapePointer).filter(isTruthy);
}
```

File: .changeset/soft-planets-watch.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": patch
+"@redocly/cli": patch
+---
+
+Fixed an issue where `$ref`s ending in `#` (instead of `#/`) would break the application.
```
Comment (lornajane) [inline .changeset/soft-planets-watch.md:6]: Do we silently handle this situation, or provide a better error message when we see it?

**Review by HCloward — APPROVED** [review]

**Review by tatomyr — APPROVED** [review]
LGTM. Thanks again @maininformer! Your changes will become available in the next release.

**Pull request merged**